apps/trading/components/swap/swap-form.tsx

```diff
@@ -0,0 +1,386 @@
+            setQuoteAsset(asset);
+          }}
+          balance={quoteAssetBalance}
+          accountAssetIds={accounts?.map((a) => a.asset.id)}
+          assets={spotAssets}
+          pubKey={pubKey}
+          testId="you-pay"
+        />
+        <SwapButton onClick={handleSwapAssets} data-testid="swap-button" />
+        <AssetInput
+          label={t('You receive')}
+          amount={baseAmount || ''}
+          asset={baseAsset}
+          balance={baseAssetBalance}
+          accountAssetIds={accounts?.map((a) => a.asset.id)}
+          assets={spotAssets}
+          onAssetChange={(asset) => {
+            setBaseAsset(asset);
+          }}
+          onAmountChange={(e) => {
+            const baseAmount = e.target.value;
+            const quoteAmount = deriveAmount({
+              amount: baseAmount,
+              marketData,
+              market,
+              baseAsset,
+              quoteAsset,
+              userValue: 'base',
+            });
+            setQuoteAmount(quoteAmount);
+            setBaseAmount(baseAmount);
+          }}
+          accountWarning={false}
+          pubKey={pubKey}
+          testId="you-receive"
+        />
+      </div>
+      <PriceImpactInput
+        value={tolerance}
+        onValueChange={(value) => setTolerance(value)}
+        data-testid="price-impact-input"
+      />
+
+      <TradingButton
+        type="submit"
+        intent={Intent.Submit}
+        data-testid="swap-now-button"
+        size="large"
+      >
+        {t('Swap now')}
+      </TradingButton>
+      <SpotData
+        price={marketPrice}
+        market={market}
+        side={side}
+        quoteAmount={quoteAmount}
+        baseAmount={baseAmount}
+        quoteAsset={quoteAsset}
+        baseAsset={baseAsset}
+      />
+    </form>
+  );
+};
+
+/** Derive the amount to be received based on best bid/offer prices */
+const deriveAmount = ({
+  marketData,
+  amount,
+  market,
+  baseAsset,
+  quoteAsset,
+  userValue,
+}: {
+  marketData: MarketDataFieldsFragment | null;
+  amount?: string;
+  market?: MarketFieldsFragment;
+  baseAsset?: AssetFieldsFragment;
+  quoteAsset?: AssetFieldsFragment;
+  userValue: 'base' | 'quote';
+}) => {
+  if (!market || !amount) return '';
+
+  const side = deriveSide({ market, baseAsset, quoteAsset, userValue });
+
+  if (side === Side.SIDE_SELL) {
+    const price = marketData?.bestBidPrice || '0';
+    const notionalSize = getNotionalSize(
+      price,
+      removeDecimal(amount, market.positionDecimalPlaces),
+      market.decimalPlaces,
+      market.positionDecimalPlaces,
+      market.decimalPlaces
+    );
+    return notionalSize ? addDecimal(notionalSize, market.decimalPlaces) : '';
+  }
+
+  if (side === Side.SIDE_BUY) {
+    const price = toBigNum(
+      marketData?.bestOfferPrice || '0',
+      market.decimalPlaces
+    );
+    return new BigNumber(amount).dividedBy(price).toString();
+  }
+
+  return '';
+};
+
+const deriveSide = ({
```

Comment:
This is also confusing, the side of the order should only be changes via the "arrow" button.

Reply:
yes, but it's not just that. 
the is also the initial state where the user selects the assets and then we need to find out if there is a market matching those two selected assets, and then we need to derive if it is a buy or a sell.

Reply:
but the side changes only via the arrow button, and in an initial version we were keeping state